src/doom/r_segs.c

```diff
@@ -194,8 +194,18 @@ R_RenderMaskedSegRange
 //  textures.
 // CALLED: CORE LOOPING ROUTINE.
 //
-#define HEIGHTBITS		12
-#define HEIGHTUNIT		(1<<HEIGHTBITS)
+//#define HEIGHTBITS		12
+//#define HEIGHTUNIT		(1<<HEIGHTBITS)
+
+// Doom 1.2 uses 16.16 fixed point numbers here
+int HEIGHTBITS = 16;
```

Comment:
Please rename these to lower case since they're not constants; also change to static globals. You can uncomment the #defines above and change this to eg.

    static int heightbits = HEIGHTBITS;
    static fixed_t heightunit = HEIGHTUNIT;